### PR TITLE
2.16.4 Release notes

### PR DESCRIPTION
**Pull request by zhen0**
Release notes for 2.16.4

### Checklist

- [x] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `maintenance`, `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [ ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed.

For new functions or classes in the Python SDK:

- [ ] This pull request includes helpful docstrings.
- [ ] If a new Python file was added, this pull request contains a stub page in the Python SDK docs and an entry in `mkdocs.yml` navigation.

### CODE REVIEW

**Review by serinamarie — COMMENTED** [review]
It doesn't look like `python scripts/generate-release-notes.py` was used for these. If you take a look at previous release notes, they have a different format that doesn't list who the PR is authored by, which I think makes it a lot less clunky/less overhead.

File: RELEASE-NOTES.md [hunk 1]
```diff
@@ -1,5 +1,63 @@
...
+
+A year ago, we released workers as a replacement for agents. Workers significantly enhance the experience of deploying flows and simplify the specification of each flow's infrastructure and runtime environment.
+
+With this release we are adding a six month (September 14) deprecation warning to agents and related concepts. Please note that:
+
+- Deprecation will not impact or break any work running with agents and agent-related concepts - although we encourage users to upgrade to workers because they provide a better deployment experience, you can continue to use existing agents and related concepts after deprecation
+- After September 14, Prefect Cloud users will not be able to create new agent work pools or infrastructure blocks
+- After September 14, new minor versions of the Prefect Python package will not include agents
+
+Like agents, workers support creating deployments through the Prefect CLI and through Python, but require different syntax.
```
Comment (desertaxle) [inline RELEASE-NOTES.md:21]: Including a link to the upgrade guide will probably help some users.

```suggestion
Like agents, workers support creating deployments through the Prefect CLI and through Python, but require different syntax.

For more information, please refer to the [Upgrade from Agents to Workers](https://docs.prefect.io/latest/guides/upgrade-guide-agents-to-workers/) guide
```

Comment (zhen0) [reply RELEASE-NOTES.md:21]: Nice!  Thank you

**Review by abrookins — APPROVED** [review]

File: RELEASE-NOTES.md [hunk 2]
```diff
@@ -1,5 +1,65 @@
...
+
+For more information, please refer to the [Upgrade from Agents to Workers](https://docs.prefect.io/latest/guides/upgrade-guide-agents-to-workers/) guide
```
Comment (billpalombi) [inline RELEASE-NOTES.md:23]:
```suggestion
For more information, please refer to the [Upgrade from Agents to Workers](https://docs.prefect.io/latest/guides/upgrade-guide-agents-to-workers/) guide.
```

File: RELEASE-NOTES.md [hunk 3]
```diff
@@ -1,5 +1,65 @@
...
+- After September 14, Prefect Cloud users will not be able to create new agent work pools or infrastructure blocks
+- After September 14, new minor versions of the Prefect Python package will not include agents
+
+Like agents, workers support creating deployments through the Prefect CLI and through Python, but require different syntax.
+
+For more information, please refer to the [Upgrade from Agents to Workers](https://docs.prefect.io/latest/guides/upgrade-guide-agents-to-workers/) guide.
+
+### Enhancements
+* Give better client-side feedback on websocket authT/authZ issues - https://github.com/PrefectHQ/prefect/pull/12221
+* Allow table artifact cells to render markdown content [`#2190`](https://github.com/PrefectHQ/prefect-ui-library/pull/2190)
```
Comment (billpalombi) [inline RELEASE-NOTES.md:27]:
```suggestion
* Allow table artifact cells to render markdown content - [`#2190`](https://github.com/PrefectHQ/prefect-ui-library/pull/2190)
```

File: RELEASE-NOTES.md [hunk 4]
```diff
@@ -1,5 +1,65 @@
...
+* Add an 'AzureBlobStorageContainer' block [`#139`](https://github.com/PrefectHQ/prefect-azure/pull/139)
```
Comment (billpalombi) [inline RELEASE-NOTES.md:28]:
```suggestion
* Add an 'AzureBlobStorageContainer' block - [`#139`](https://github.com/PrefectHQ/prefect-azure/pull/139)
```

File: RELEASE-NOTES.md [hunk 5]
```diff
@@ -1,5 +1,65 @@
...
+* Deprecate `Deployment` class and deployment `build` and `apply` commands - https://github.com/PrefectHQ/prefect/pull/12283
+* Deprecate the `packaging` module - https://github.com/PrefectHQ/prefect/pull/12291
+* Deprecate storage blocks with implementations in Prefect integration packages - https://github.com/PrefectHQ/prefect/pull/12299
+* Do not create `default-agent-pool` in empty state server - https://github.com/PrefectHQ/prefect/pull/12214
+### Typing
+* Refactor: Inject Client - https://github.com/PrefectHQ/prefect/pull/12258
+* Fix: Variables methods type-hinting - https://github.com/PrefectHQ/prefect/pull/12278
+### Documentation
+* Fix formatting in concept -> work-pools.md - https://github.com/PrefectHQ/prefect/pull/12230
+* Docs: Add `get_run_logger` to imports in interactive workflow examples - https://github.com/PrefectHQ/prefect/pull/12284
```
Comment (billpalombi) [inline RELEASE-NOTES.md:52]:
```suggestion
* Add `get_run_logger` to imports in interactive workflow examples - https://github.com/PrefectHQ/prefect/pull/12284
```

**Review by billpalombi — APPROVED** [review]
lgtm

**Pull request merged**